services/web/server/src/simcore_service_webserver/projects/_nodes_handlers.py

```diff
@@ -469,23 +470,55 @@ async def get_project_services_access_for_gid(request: web.Request) -> web.Respo
         ]
     )
 
+    # Initialize groups to compare with everyone group ID
+    groups_to_compare = {EVERYONE_GROUP_ID}
+
+    # Get the group from the provided group ID
+    _sharing_with_group: Group | None = await get_group_from_gid(
+        app=request.app, gid=query_params.for_gid
+    )
+
+    # Check if the group exists
+    if _sharing_with_group is None:
+        msg = f"Group ID does not exist: {query_params.for_gid}"
+        raise ValueError(msg)
+
+    # Update groups to compare based on the type of sharing group
+    if _sharing_with_group.group_type == GroupTypeInModel.PRIMARY:
+        _user_id = await get_user_id_from_gid(
+            app=request.app, primary_gid=query_params.for_gid
+        )
+        _, _user_groups, _ = await list_user_groups(app=request.app, user_id=_user_id)
+        groups_to_compare.update({int(item.get("gid")) for item in _user_groups})
+        groups_to_compare.add(query_params.for_gid)
```

Comment:
probaby `for_gid` this is already in the `set` but ok :-)

Reply:
It is not. I was also surprised when working with it the first time somehow I was expecting list_user_groups to get a list of all groups. But the response is a tuple (dict, list[dict], dict) where the first part is the primary group, the second one is all his organizations and the third one is everyone group. 

Therefore primary group is not part of the set you are referencing.

Reply:
well, if a function called `list_user_groups` does not return the primary group id I would say it has a design flaw